apps/gitness/src/pages-v2/repo/repo-code.tsx

```diff
@@ -0,0 +1,148 @@
+import { useEffect, useMemo, useState } from 'react'
+import { useParams } from 'react-router-dom'
+
+import {
+  GitPathDetails,
+  OpenapiContentInfo,
+  OpenapiGetContentOutput,
+  pathDetails,
+  useFindRepositoryQuery,
+  useGetContentQuery
+} from '@harnessio/code-service-client'
+import { RepoFile, RepoFiles, SummaryItemType } from '@harnessio/ui/views'
+
+import { useGetRepoRef } from '../../framework/hooks/useGetRepoPath'
+import { timeAgoFromISOTime } from '../../pages/pipeline-edit/utils/time-utils'
+import { PathParams } from '../../RouteDefinitions'
+import { getTrimmedSha, normalizeGitRef } from '../../utils/git-utils'
+import { splitPathWithParents } from '../../utils/path-utils'
+
+/**
+ * TODO: This code was migrated from V2.
+ * @constructor
+ */
+export const RepoCode = () => {
+  const repoRef = useGetRepoRef()
+  const { spaceId, repoId, gitRef, resourcePath } = useParams<PathParams>()
+  const subResourcePath = useParams()['*'] || ''
+  const repoPath = `/${spaceId}/repos/${repoId}/code/${gitRef}`
+  const fullResourcePath = subResourcePath ? resourcePath + '/' + subResourcePath : resourcePath
+  // TODO: pathParts - should have all data for files path breadcrumbs
+  const pathParts = [
+    {
+      path: repoId!,
+      parentPath: repoPath
+    },
+    ...splitPathWithParents(fullResourcePath || '')
+  ]
+  const [files, setFiles] = useState<RepoFile[]>([])
+  const [loading, setLoading] = useState(false)
+  const [selectedBranch, setSelectedBranch] = useState<string>(gitRef || '')
+
+  const { data: { body: repoDetails } = {} } = useGetContentQuery({
+    path: fullResourcePath || '',
+    repo_ref: repoRef,
+    queryParams: { include_commit: true, git_ref: normalizeGitRef(gitRef || '') }
+  })
+  const { data: { body: repository } = {} } = useFindRepositoryQuery({ repo_ref: repoRef })
+
+  useEffect(() => {
+    if (repository && !gitRef) {
+      setSelectedBranch(repository?.default_branch || '')
+    } else if (gitRef) {
+      setSelectedBranch(gitRef)
+    }
+  }, [repository, gitRef])
+
+  const repoEntryPathToFileTypeMap = useMemo((): Map<string, OpenapiGetContentOutput['type']> => {
+    if (repoDetails?.content?.entries?.length === 0) return new Map()
```

Comment:
This will not return a new `Map` if `repoDetails`, `content` or `entries` are not present. Checking for falseyness will account for that
```suggestion
    if (!repoDetails?.content?.entries?.length) return new Map()
```